components/tensorflow-notebook-image/Dockerfile

```diff
@@ -124,7 +125,15 @@ RUN conda create -n py2 python=2 && \
     python -m ipykernel install && \
     # tensorflow-model-analysis is only supported for TF 1.6 and above
     if [[ $INSTALL_TFMA == "yes" ]]; then \
-      pip install --no-cache-dir tensorflow-model-analysis && \
+      # if you find any dependency problems in the future
+      # please visit https://github.com/tensorflow/model-analysis#compatible-versions
+      # and fix the TFMA_VERSION with compatible version number in the config json
+      if [[ $TFMA_VERSION == "default" ]]; then \
```

Comment:
Can we simplify this 

1) Why do we need to support "default"
2) Get rid of INSTALL_TFMA and just rely on TFMA_VERSION.

Reply:
1. For example, if the tfma has a new version with for  tf==1.9 and we specified the tfma version, now we are possible to ignore the tfma update. But if we have the default, we will update the tfma as the pip update. And until the tfma does not support the version in the latest release, such as tf < 1,9, we specified its version.
2. I will combine this two and use TFMA_VERSION to 'disbale_install' when tf < 1.6. I haven't combine the two at first because i don't want to the TFMA_VERSION has the other meaning, it hurt the concise of code.

Reply:
I'm not sure we want to auto-update. That could break things. I think we are moving in the direction of pinning things.

See for example #1751 

I would suggest removing "default". If TFMA releases a new version for existing TF versions then we can explicitly upgrade.

Reply:
Ok, it's a better way if we have the auto way to modify the pinning version.  I will remove it.